packages/flutter/lib/src/widgets/app.dart

```diff
@@ -234,6 +234,15 @@ typedef PageRouteFactory = PageRoute<T> Function<T>(RouteSettings settings, Widg
 /// Creates a series of one or more initial routes.
 typedef InitialRouteListFactory = List<Route<dynamic>> Function(String initialRoute);
 
+/// Disposes navigation related singletons.
+@visibleForTesting
+void disposeFlutterSingletons() {
+  // ignore: invalid_use_of_visible_for_testing_member, https://github.com/dart-lang/sdk/issues/41998
+  disposeNavigationSingletons();
+  WidgetsApp._debugShowWidgetInspectorOverrideNotifierObject?.dispose();
+  WidgetsApp._debugShowWidgetInspectorOverrideNotifierObject = null;
+}
+
```

Comment:
What is the suggested pattern for when we add more singletons like that in other parts of the framework? Do they need to be added to this one method here to dispose them all by calling `disposeFlutterSingletons`? Please document that.

What are package authors (or app authors) supposed to do that use similar singletons in their package/app? How do they hock into this system to dispose their singleton? Is this documented somewhere?